pkg/sentry/sentry.go

```diff
@@ -34,6 +35,44 @@ type SentryErrorResponse struct {
 	Detail string `json:"detail"`
 }
 
+func (sc *SentryClient) FetchWithPagination(path string, out interface{}) (string, error) {
+	fullURL := path
+	if !strings.HasPrefix(path, sc.BaseURL) {
+		fullURL = sc.BaseURL + path
+	}	
+	req, _ := http.NewRequest(http.MethodGet, fullURL, nil)
+	res, err := sc.sentryHttpClient.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer res.Body.Close()
+
+	nextURL := ""
+	header := res.Header
+	links := link.ParseHeader(header)
+
+	if links != nil {
+		if nextLink, found := links["next"]; found && nextLink.Extra["results"] == "true" {
+			nextURL = nextLink.URI
+		}
+	}
+
+	if res.StatusCode == http.StatusOK {
+		if err := json.NewDecoder(res.Body).Decode(&out); err != nil {
+			return "", err
+		}
+	} else {
+		var errResponse SentryErrorResponse
+		if err := json.NewDecoder(res.Body).Decode(&errResponse); err != nil {
+			errorMessage := strings.TrimSpace(fmt.Sprintf("%s %s", res.Status, err.Error()))
+			return "", errors.New(errorMessage)
+		}
+		errorMessage := strings.TrimSpace(fmt.Sprintf("%s %s", res.Status, errResponse.Detail))
+		return "", errors.New(errorMessage)
```

Comment:
It would be nice to move this part to
```
func (err SentryErrorResponse) Error() string {
     // ...
}
```
and in here simply `return "", errResponse`

Reply:
I'm not sure I understand what you mean. Could you please provide more details?

Reply:
I wanted to suggest something like:

```Go
type SentryErrorResponse struct {
	Detail string `json:"detail"`
	decodingErr error
	responseStatus string
}

func (err *SentryErrorResponse) Error() string {
	if err.decodingErr != nil {
		return strings.TrimSpace(fmt.Sprintf("%s %s", res.Status, err.Error()))
	}
	return strings.TrimSpace(fmt.Sprintf("%s %s", res.Status, errResponse.Detail))
}

func responseError(res *http.Response) error {
	err := &SentryErrorResponse{
		responseStatus res.Status
	}
	err.decodingErr = json.NewDecoder(res.Body).Decode(err)
	return err
}

func (sc *SentryClient) FetchWithPagination(path string, out interface{}) (string, error) {
	// ...
	if res.StatusCode == http.StatusOK {
		// ...
	} else {
		return "", responseError(res)
	}
```

But then I realised `SentryErrorResponse` is also used in the `Fetch()` method. So ignore me :-)